tests/Cache/CacheTest.php

```diff
@@ -148,32 +150,48 @@ public function providerMissingData()
         }, array_keys($data));
     }
 
-    public function testCanConvertToAndFromJson()
+    /**
+     * @dataProvider provideCanConvertToAndFromJsonCases
+     */
+    public function testCanConvertToAndFromJson(SignatureInterface $signature)
     {
-        $signature = new Signature(
-            PHP_VERSION,
-            '2.0',
-            array(
-                'foo',
-                'bar',
-            )
-        );
-
         $cache = new Cache($signature);
 
         $file = 'test.php';
         $hash = crc32('hello');
 
         $cache->set($file, $hash);
 
-        /* @var Cache $cached */
+        /** @var Cache $cached */
```

Comment:
maybe this type hint can be removed as a whole as the method `Cache::fromJson` already type hints the return value correctly?